### PR TITLE
docs: add Liferay.component, Liferay.componentReady, Liferay.destroyComponent and Liferay.destroyComponents

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

File: dxp/liferay_component.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+When the component is registered a global Liferay event `id + :registered` is fired, which can be listen with `Liferay.on()` o `Liferay.once()`.
+
+⚠️ WARNING: If there's already a component registered with the passed `id` it will be replaced with the new value, which could lead to unexpected behaviour in the "Liferay.component" and "Liferay.componentReady" APIs, as well as in the `*:registered` events.
+
+### Retrieve
+
+`Liferay.component(id)`
+
+Once the component is registered it can be retrieved by passing to the method the id it was registered with.
```
Comment (wincent) [inline dxp/liferay_component.md:23]: You can reword to not end sentences with a preposition (with).

> A previously registered component can be retrieved by passing the same `id`.

File: dxp/liferay_component.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+One thing to take into account is that if the registered value was a function the first time the component is retrieved the function will be executed and its result stored as the new registered value, which will be returned.
```
Comment (wincent) [inline dxp/liferay_component.md:25]: Comma after "value was a function".

File: dxp/liferay_component.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+### Notes
+
+Using `<react:component>` tag a component will be registered using `Liferay.Component` but it will not be the react component but a synthetic one, that's why even when a function component is being registered (which doesn't create an instance) a component instance is returned.
```
Comment (wincent) [inline dxp/liferay_component.md:29]: "Using" → "Using the"

"react" → "React"

"synthetic one, that's why" → "synthetic one. That's why"

File: dxp/liferay_component.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+# Liferay.componentReady()
```
Comment (wincent) [inline dxp/liferay_component.md:31]: Backticks

File: dxp/liferay_component.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+`Liferay.componentReady()` retrieves a list of component instances after they've been registered and returns a `Promise` to be resolved with all the requested component instances after they've been successfully registered.
+
+## How to use it
+
+`Liferay.componentReady([...id]).then(function)`
```
Comment (wincent) [inline dxp/liferay_component.md:37]: Fenced codeblock?

File: dxp/liferay_component.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+A common use case is the need of run some actions after a component which is not in scope is created.
+
+# Liferay.destroyComponent()
```
Comment (wincent) [inline dxp/liferay_component.md:41]: Backticks.

File: dxp/liferay_component.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+This is the safest way to destroy a registered component. Destroys the component registered by the provided component ID. This invokes the component's own destroy lifecycle methods (destroy or dispose) and deletes the internal references to the component in the component registry.
```
Comment (wincent) [inline dxp/liferay_component.md:43]: "Destroys" → "It destroys"

"registered by" → "registered with"

Backticks around `destroy` and `dispose`

File: dxp/liferay_component.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+`Liferay.destroyComponent(componentId)`
```
Comment (wincent) [inline dxp/liferay_component.md:45]: Fenced?

File: dxp/liferay_component.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+# Liferay.destroyComponents()
```
Comment (wincent) [inline dxp/liferay_component.md:47]: Backticks

File: dxp/liferay_component.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+Destroys registered components matching the provided filter function. If no filter function is provided, it destroys all registered components.
+
+`Liferay.destroyComponents(filterFn)`
```
Comment (wincent) [inline dxp/liferay_component.md:51]: Fence.

File: dxp/liferay_component.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+`filterFn` is a method that receives a component's destroy options and the component itself, and returns <code>true</code> if the component should be destroyed.
```
Comment (wincent) [inline dxp/liferay_component.md:53]: Why `<code>` here instead of just backticks?

**Review by wincent — COMMENTED** [review]
I only have nits; this looks good to me.

File: dxp/liferay_component.md (hunk 11 above)
Comment (carloslancha) [reply dxp/liferay_component.md:53]: Copy/pasted from js docs 😅

Comment (wincent) [reply dxp/liferay_component.md:53]: copy-pasta

**Comment by carloslancha** [comment]
Feedback applied!

File: dxp/liferay_component.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+### Retrieve
+
+```js
+Liferay.component(id);
+```
+
+A previously registered component can be retrieved by passing the same `id`.
+
+One thing to take into account is that if the registered value was a function, the first time the component is retrieved the function will be executed and its result stored as the new registered value, which will be returned.
```
Comment (wincent) [inline dxp/liferay_component.md:29]: Crafty little devil.

crafty-little-devil

**Review by wincent — APPROVED** [review]
lg2m

**Comment by jbalsas** [comment]
Hey @carloslancha, the document looks good to me!

I forgot to mention here that I think it would be interesting to document some of the ootb lifecycles. 

We are destroying components in various places:
- Portlet destroy
- SPA navigation

By default we're also registering components in different ways (in tags with the `portletId` in other cases with `destroyOnNavigate`...). This makes developers not really have to worry about the `destroyComponent` API directly but rather rely on the system to do the proper cleanup.

I'd prefer to see this explained in detail to show how to properly register a component and why and hide the `destroy*` APIs a little bit so that we don't give the impression that those are APIs meant to be consumed by every product team in their developments.

Can you follow-up with some of these on top?

Thanks! 🙏

**Comment by carloslancha** [comment]
@jbalsas https://github.com/liferay/liferay-frontend-guidelines/pull/208

**Pull request merged**